server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -433,6 +594,39 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field(CODEC_VERSION_FIELD.getPreferredName(), getCodecVersion());
             builder.field(GLOBAL_METADATA_FIELD.getPreferredName(), getGlobalMetadataFileName());
         }
+        if (onOrAfterCodecVersion(CODEC_V3)) {
+            if (getDiscoveryNodesMetadata() != null) {
```

Comment:
When do we expect this to be null for CODEC_V3?

Reply:
If remote publication is turned off, we are not changing the codec back to V2. The codec will remain same so we need to make sure that we can work with publication turned off

Reply:
We are adding empty blocks in that case right?

Reply:
We will not even add attributes like DiscoveryNodes or ClusterBlocks or DiffManifest to XContent. Not sure what you mean by empty block.

Reply:
Got it. I was assuming we are adding empty params when data is not there. If thats not the case, then the check makes sense.

Reply:
@shiv0408 : To understand better, if remote publication is turned off, the ephemeral objects will not be stored in S3 and subsequently not used for publication as well.

Reply:
I think you will need a flag in manifest to capture what is uploaded otherwise, when the setting is being turned on and off, it may cause issues.

Reply:
@shwetathareja Yes, if publication is turned off, no ephemeral objects in remote store. What kind of issues do you see as we have written  parsers in a way that parsing will not break if objects are not present in XContent. If publication is turned on manifest will have those values, we just need to careful while accessing those objects to ensure that they are non null.